server/protocols/jmap-draft/src/main/java/org/apache/james/jmap/draft/methods/MessageAppender.java

```diff
@@ -176,35 +170,29 @@ private Flags getFlags(CreationMessage message) {
     }
 
     private ImmutableList<MessageAttachmentMetadata> getMessageAttachments(MailboxSession session, ImmutableList<Attachment> attachments) throws MailboxException {
-        Map<AttachmentId, AttachmentMetadata> attachmentsById = attachmentManager.getAttachments(attachments.stream()
-            .map(attachment -> AttachmentId.from(attachment.getBlobId().getRawValue()))
-            .collect(ImmutableList.toImmutableList()), session)
+        return attachments
             .stream()
-            .collect(ImmutableMap.toImmutableMap(AttachmentMetadata::getAttachmentId, Function.identity()));
-
-        ThrowingFunction<Attachment, Optional<MessageAttachmentMetadata>> toMessageAttachment = att -> messageAttachment(att, attachmentsById);
-
-
-        return attachments.stream()
-            .map(Throwing.function(toMessageAttachment).sneakyThrow())
-            .flatMap(Optional::stream)
+            .flatMap(attachment -> {
+                try {
+                    Blob blob = blobManager.retrieve(attachment.getBlobId(), session);
+                    return Stream.of(MessageAttachmentMetadata.builder()
+                        .attachment(AttachmentMetadata.builder()
+                            .attachmentId(AttachmentId.from(blob.getBlobId().getRawValue()))
```

Comment:
blob.getBlobId().asAttachmentId()

Reply:
Solved in https://github.com/apache/james-project/pull/584 (upcoming)